app/controllers/dependencies_controller.rb

```diff
@@ -29,41 +19,24 @@ def edit
   def update
     @dependency = Dependency.find(params[:id])
 
-    respond_to do |format|
-      if @dependency.update_attributes(dependency_params)
-        format.html { redirect_to @dependency, :notice => 'Dependency was successfully updated.' }
-        format.json { head :ok }
-      else
-        format.html {
-          render :action => "edit"
-        }
-        format.json { render :json => @dependency.errors, :status => :unprocessable_entity }
-      end
+    if @dependency.update_attributes(dependency_params)
```

Comment:
how about using respond_with? I think the json responses are quite useful here, as they potentially allow external scripts to hook into labs.